xmppserver/src/main/java/org/jivesoftware/openfire/container/PluginMonitor.java

```diff
@@ -286,7 +272,10 @@ public boolean accept( final Path path ) throws IOException
                             // If the JAR needs to be exploded, do so.
                             if ( Files.notExists( dir ) )
                             {
-                                unzipPlugin( canonicalPluginName, jarFile, dir );
+                                if (!unzipPlugin( canonicalPluginName, jarFile, dir ) )
+                                {
+                                    return;
```

Comment:
Does this want to be a continue, so that later in the runnable plugins can still be unloaded, loaded, etc?

Reply:
oh, good catch!